### ISSUE TITLE
width/height 0 and transform + overflow hidden

### ISSUE THREAD

**Issue by sainthkh** [cypress-io/cypress#6093]
### Current behavior:

When an element has width or height 0px with non-`none` transform and overflow `hidden`, it is not visible. But `isHidden()` cannot detect it. 

```html
<!-- Example -->
<div style="height: 0px; transform: translate(1, 2)"><p>Visible</p></div>
<div style="height: 0px; transform: translate(1, 2); overflow: hidden"><p>Invisible</p></div>
```

It is found with cypress-example-piechopper in https://circleci.com/gh/cypress-io/cypress-test-example-repos/71863. 

### Desired behavior:

`isHidden()` should detect it. 

### Steps to reproduce: (app code and test code)

1. Download https://github.com/cypress-io/cypress-example-piechopper.
2. Change cypress to dev version. 
3. Run app_spec.js 
4. "displays hamburger menu" fails.

### Versions

After #6000 is merged.

**Comment by sainthkh** [comment]
Will be fixed with #5916.

**Comment by jennifer-shehane** [comment]
Released in [3.8.2](https://on.cypress.io/changelog#3-8-2)
